(zotero) add 64-bit installer option (re-re-try)

teknowledgist opened the pull request:
Third-time is the charm!  😄 

This adds the new install option for 64-bit Zotero and simplifies the update a bit.  This is essentially [the same PR as submitted -- and closed -- previously](https://github.com/chocolatey-community/chocolatey-packages/pull/2546) but the version has been updated and it is hopefully "cleaner".

## Description

* Modified the `update.ps1` to grab both 32-bit and 64-bit installer URLs, download the EXEs and replace the information in the `VERIFICATION.txt` file. 
* Modified `chocolateyinstall.ps1` to install either 32-bit or 64-bit flavor of Zotero. 
* Adjusted the `VERIFICATION.txt file` to better match the "standards" already in the repo.  
* Added to `chocolateyinstall.ps1` and `chocolateyuninstall.ps1` files to work for both bit levels. 
* Updated and re-arranged metadata in the .nuspec to match the order of a new package. 
* Updated the LICENSE.txt file to match changes at the project.

## Motivation and Context

This addresses [issue 2531](https://github.com/chocolatey-community/chocolatey-packages/issues/2531)

32-bit Zotero works fine on 64-bit systems, but regularly pops up a dialog indicating that it's not the most efficient version for the environment. The upgrade from 32-bit to 64-bit is as simple as a re-install, so fixing this package to have both bit-levels was fairly straightforward.

## How Has this Been Tested?
In Windows 11 and the official test environment.  Installed the 32-bit-only, v7.0.9 package then upgraded to the local, modified v7.0.11 package from this PR with no trouble. Also installed the proposed package without upgrading. Uninstalls worked too.

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:
- [x] My pull request is not coming from the master branch.
- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [ ] I have updated the package description and it is less than 4000 characters.
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md)
- [x] The added/modified package passed install/uninstall in the Chocolatey Test Environment(https://github.com/chocolatey-community/chocolatey-test-environment/). _Note that we don't support the use of any other environment_.
- [x] The changes only affect a single package (not including meta package).

ScottMonolith commented:
@pauby / @TheCakeIsNaOH - anything else needed here or can this be merged?  I'm not sure what this error is indicating.

`Error: Missing either 'issue-number' or 'comment-id'`

#2531 is the issue number.

traits commented:
Zotero is now v7.0.15. The extent of formal red tape in OpenSource these days is absolutely staggering.

pauby commented:
> All that said, this PR isn't the appropriate place to have that discussion.

Like @teknowledgist, I don't want to add to the noise in this PR. However, I think it's important enough to say that _everything_ you said is correct, and we struggle with the balance here. Getting community feedback is like herding cats, so I would be delighted for somebody to open a [discussion](https://github.com/chocolatey-community/chocolatey-packages/discussions) on this if they want to provide constructive feedback and suggestions.

teknowledgist commented:
@pauby Done.  https://github.com/orgs/chocolatey-community/discussions/2652

pauby reviewed (commented):
A couple of questions as to why some changes were made.

teknowledgist commented on automatic/zotero/legal/VERIFICATION.txt:
```diff
@@ -2,17 +2,22 @@ VERIFICATION
   - Use chocolatey utility 'checksum.exe'
 3. The checksums should match the following:
 
-  checksum type: sha256
```
TBH, I don't know.  It was probably inadvertent.

I do see that about 1/3 of the packages with a `VERIFICATION.txt` file do not indicate what the checksum type is.

pauby replied:
Can you add it back in? Removing it was unnecessary.

teknowledgist replied:
I already did last week in the "requested tweaks" change. 😄

teknowledgist commented on automatic/zotero/update.ps1:
```diff
@@ -1,42 +1,36 @@
-      "(?i)^(\s*softwareName\s*=\s*)'.*'" = "`${1}'$softwareName'"
-    }
-    ".\$($Latest.PackageName).nuspec" = @{
-      "(\<releaseNotes\>).*(\<\/releaseNotes\>)" = "`${1}https://www.zotero.org/support/$($version.Major).$($version.Minor)_changelog`${2}"
```
Because it was one more thing that could potentially change and to maintain.  

Also, the URL for the latest major.minor version is redirected anyway to just https://www.zotero.org/support/changelog (what is in the `.nuspec` statically now).  The user would probably need to scroll to find their specific version anyway, and there is a link at the bottom to access the older version.

pauby replied:
If you are going to take responsibility for maintaining the Zotero package in this repository, I'm happy to let this change go through. 

If you are going to be responsible, please also add a change to the CODEOWNERS file to add your name and the zotero package,

If you're not, then making changes that don't need to be made causes issues for others that would be maintaining the package, so I would ask that you only make changes that get you to the goal of this PR: 'add 64-bit installer option'.

teknowledgist replied:
Hrmmm.  What does it mean to "take responsibility" *in this repository*? What benefit is there for me (or anyone) to that over an individual (like myself) just being the maintainer and hosting the package code in their own repository? I don't know about other package maintainers, but had I been the maintainer of Zotero, it would have had the 64-bit installer back in late August.

Also, if I strictly only make changes related to adding the 64-bit installer, there are a number of other changes that will have to be removed (like the change in author, license path, etc.). I think there is a fair amount of grey area here. Is a static URL that matches what the dynamic URL redirects to anyway really a problem?  It seems like a more stable state to me.

pauby replied:
>  What does it mean to "take responsibility" in this repository?

It means that if any PR comes in for the zotero package, you would automatically be added as the reviewer on the package and be "responsible" for reviewing those PR's. It is hoped, but not required, that you would be proactive in keeping the package running.

> What benefit is there for me (or anyone) to that over an individual (like myself) just being the maintainer and hosting the package code in their own repository?

You would be solely responsible for it in your own package repository without the help of the additional community that we have built, and are currently trying to expand upon.

> Also, if I strictly only make changes related to adding the 64-bit installer, there are a number of other changes that will have to be removed (like the change in author, license path, etc.). I think there is a fair amount of grey area here.

For me (and I'm just using me as an example) to quickly review these changes, sticking to "adding 64 bit installer" in your changes makes that easier. If you then want to make other changes that are not related to that, there is more for me to review and, more importantly, more for me to consider. I am not an expert on zotero, so I then need to understand more about your changes.

> Is a static URL that matches what the dynamic URL redirects to anyway really a problem? _It seems like a more stable state to me._

(the emphasis there is mine).

You have the knowledge and the experience to determine if that dynamic URL is more stable than the static one. I don't have that. So I have to ask myself if it is, and why it is, and then I need to answer that question myself, which takes time. And as it's not related to "adding 64 bit installer" I'm left wondering why it's been added at all in this PR and why not a separate PR.

It's these changes, and these questions, that otherwise hold up a simple PR that is just "adding 64 bit installer". If you break the changes out into separate PR's we can take our time with the _other_ changes you want to make, ask the questions and answer the questions without holding up important changes. This is how the PR systems work.

You asked a question about a lot of this over in Discussions which I have held back from answering as @gep13 is going to do that (this week).

teknowledgist commented on automatic/zotero/update.ps1:
```diff
@@ -1,42 +1,36 @@
 function global:au_GetLatest {
-    $url = Get-RedirectedUrl -url $releases
+   $url64 = Get-RedirectedUrl -url $releases
+   $url32 = $url64 -replace 'x64','win32'
```
OK.  adjusted.  Hopefully it doesn't break this PR like before.

b-a0 commented:
@teknowledgist @pauby is there perhaps an update on this pull request? From the comments/labels it's not clear to me what needs to happen to get this merged. Thanks in advance for your reply!

teknowledgist commented:
It's not clear to me either.

corbob commented:
> It's not clear to me either.

Not sure if it shows up different because I can merge commits, but as it stands, there's merge conflicts that git/GitHub can't resolve on it's own. At the very least the merge conflict will need to be resolved.

<img width="855" height="375" alt="image" src="https://github.com/user-attachments/assets/836e93d1-07e8-4dbc-816b-d38db2b6525f" />

teknowledgist commented:
So to be very humble and honest, I don't understand the conflicts, and I am certain it is because I have huge gaps in my understanding of Git/GitHub.  I am pretty sure those conflicts weren't there when I submitted the PR, but appeared later when the version of Zotero was updated.

I (guess I) understand Diffs, but conflicts are clearly not that.  It would make sense to me if every line in the Chocolatey-Community Master files that changed after I submitted my PR was marked as a conflict. It would essentially be saying that there is a difference in the Diffs (one between the old version and my changes and the other between the new version and my changes).  I don't know how that would get fixed, but the concept would make sense.  

Instead, the conflicts show some lines that are different between when I submitted the PR and the current files, but don't seem to care about other lines that have also changed due to the newer version.  

One of the previous (failed) times I submitted this PR, I thought the conflicts were because the repository updated the package to a newer version than against what I had submitted.  Thus, I updated my copy of the package and updated the PR so the version (of Zotero) was the same (with no other changes).  I believe doing that was ultimately what caused the failure of the PR.  I was told (somewhere) to NOT change the PR. I don't know how I can't change the PR, AND hit the moving target that is the auto-updating repository.  

I don't know what "Resolve conflicts" is showing me, nor what I am supposed to do (or can do) to actually resolve it.  

I apologize (again) for being stupid about this.  I am sure it is frustrating on your end.  It is very frustrating on my end when I have the knowledge and take the time to "offer my good citizen contribution" to fix a small problem in a particular way, but that way -- which would be perfectly functional for personal/small scale repository -- is not acceptable for a large scale project for reasons I don't understand.  It's not your job to explain why, so I don't expect that either.

Thanks.
